controllers/provider-gcp/charts/internal/gcp-infra/templates/main.tf

```diff
@@ -121,6 +154,24 @@ output "{{ .Values.outputKeys.vpcName }}" {
   value = "{{ required "vpc.name is required" .Values.vpc.name }}"
 }
 
+{{ if .Values.create.cloudRouter -}}
+output "{{ .Values.outputKeys.cloudRouter }}" {
+  value = "${google_compute_router.router.name}"
+}
+{{- end }}
+
+{{ if  .Values.vpc.cloudRouter -}}
```

Comment:
WDYT about the following:

```terraform
{{ if or  .Values.create.cloudRouter .Values.vpc.cloudRouter -}}
output "{{ .Values.outputKeys.cloudRouter }}" {
  {{ if .Values.create.cloudRouter -}}
  value = "${google_compute_router.router.name}"
  {{ else -}}
  value = "{{ .Values.vpc.cloudRouter.name }}"
  {{ end -}}
}

output "{{ .Values.outputKeys.cloudNAT }}" {
  value = "${google_compute_router_nat.nat.name}"
}
{{- end }}
```

This makes more clear that a cloud router + NAT gateway is only available as output if either a new cloud router is created or a name of an existing cloud router is given.